diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,18 +25,20 @@
 if(PREFER_QT_5)
     set(QT_NS Qt5)
     set(DTK_NS Dtk)
+    set(ASQT_NS AppStreamQt5)
 else()
     set(QT_NS Qt6)
     set(DTK_NS Dtk6)
+    set(ASQT_NS AppStreamQt)
 endif()

 find_package(QT NAMES ${QT_NS} REQUIRED COMPONENTS Core)
 find_package(${QT_NS} REQUIRED COMPONENTS Core Gui Concurrent Qml Svg Quick QuickControls2 LinguistTools)
 find_package(${DTK_NS} REQUIRED COMPONENTS Core Gui)
+ find_package(${ASQT_NS} 1.0)

 if(${QT_VERSION_MAJOR} EQUAL "5")
     find_package(${QT_NS} REQUIRED COMPONENTS QuickCompiler)
-    find_package(AppStreamQt) # AppStreamQt only support Qt 5 at this moment
+    set_package_properties(${ASQT_NS} PROPERTIES
+        DESCRIPTION "Library that lists Appstream resources"
+        URL "https://www.freedesktop.org"
+        TYPE RECOMMENDED)
 endif()

diff --git a/desktopintegration.cpp b/desktopintegration.cpp
--- a/desktopintegration.cpp
+++ b/desktopintegration.cpp
@@ -65,7 +65,7 @@
     // qDebug() << pool.flags() << currentDE;
     pool.load();

-    const QList<AppStream::Component> components = pool.componentsByLaunchable(AppStream::Launchable::KindDesktopId, desktopId);
+    const AppStream::ComponentBox components = pool.componentsByLaunchable(AppStream::Launchable::KindDesktopId, desktopId);
     for (const AppStream::Component & component : components) {
         return component.compulsoryForDesktops().contains(currentDE);
     }

diff --git a/src/ddeintegration/CMakeLists.txt b/src/ddeintegration/CMakeLists.txt
--- a/src/ddeintegration/CMakeLists.txt
+++ b/src/ddeintegration/CMakeLists.txt
@@ -133,10 +133,10 @@
 PRIVATE
     Qt::Core Qt::Gui ${DTK_NS}::Core launcher-utils)

- if (AppStreamQt_FOUND)
+ if (${ASQT_NS}_FOUND)
     target_link_libraries(dde-integration-dbus
     PRIVATE
-        AppStreamQt)
+        ${ASQT_NS})
 else()
     target_compile_definitions(dde-integration-dbus
     PUBLIC NO_APPSTREAM_QT
